src/utils/build_url.ts

```diff
@@ -2,20 +2,18 @@ import type { ApiQueryParameters, Url } from '../types/options.js'
 
 const isBrowser = typeof location !== 'undefined' && typeof document !== 'undefined'
 
-const stringifyQuery = queryObj => new URLSearchParams(queryObj).toString()
-
-export function buildUrlFactory (instanceApiEndpoint: Url) {
+export function buildUrlFactory (instanceApiEndpoint: Url): BuildUrlFunction {
   return function (queryObj: ApiQueryParameters): Url {
     // Request CORS headers if the request is made from a browser
     // See https://www.wikidata.org/w/api.php ('origin' parameter)
     if (isBrowser) queryObj.origin = '*'
 
-    // Remove null or undefined parameters
-    Object.keys(queryObj).forEach(key => {
-      if (queryObj[key] == null) delete queryObj[key]
-    })
-
-    return instanceApiEndpoint + '?' + stringifyQuery(queryObj)
+    const queryArray = Object.entries(queryObj)
+      // Remove null or undefined parameters
+      .filter(([ _key, value ]) => typeof value !== 'undefined' && value !== null)
```

Comment:
I generally prefer the more succinct and strictly(?) equivalent `value != null` for this kind of check
```suggestion
      .filter(([ _key, value ]) => value != null)
```

Reply:
In my head `!=` is a bad practice and I also try to prefer `undefined` over `null` wherever I can so I'm not really sure.

Maybe a better check here would be to check for the actual types `string | number`:

```suggestion
      .filter(([ _key, value ]) => typeof value === 'string' || typeof value === 'number')
```

Reply:
I'm from the [Douglas Crockford school: always use strict equality, with this one exception to check if a value is null or undefined](https://stackoverflow.com/a/16253116/3324977)

Reply:
It's good to critically question some things. I like that!